test/testGraphical.m

```diff
@@ -0,0 +1,44 @@
+function [ status ] = testGraphical( varargin )
+%TESTGRAPHICAL Runs the M2T test suite to produce graphical output
+%
+% This is quite a thin wrapper around testMatlab2tikz to run the test suite to
+% produce a PDF side-by-side report.
+%
+% Its allowed arguments are the same as those of testMatlab2tikz.
+%
+% Usage:
+%
+%     status = testGraphical(...) % gives programmatical access to the data
+%
+%     testGraphical(...); % automatically invokes makeLatexReport afterwards
+%
+% See also: testMatlab2tikz, testHeadless, makeLatexReport
+
+    status = testMatlab2tikz('actionsToExecute', @actionsToExecute, ...
+                             varargin{:});
+
+    if nargout == 0
+        makeLatexReport(status);
+    end
+
+end
+
+function status = actionsToExecute(status, ipp)
+    status = execute_plot_stage(status, ipp);
+    
+    if status.skip
+        return
```

Comment:
You're 100% right: when a test is skipped, obviously no such fields are created by subsequent steps.

I don't like setting `status.XStage.error` for all `X` since that both requires knowledge of what stages X are to follow (which might change in the future). Also: the only real problem is that skip, so it is also easier to check for that (this has been implemented by now). The other solution would need changes during the actual action of skipping and that code is quite hard to factor out into a function (due to the GOTO-like nature of `return`).